index.js

```diff
@@ -308,15 +308,50 @@ SendStream.prototype.hasTrailingSlash = function hasTrailingSlash () {
 }
 
 /**
+ * Returns `true` if any request preconditions are failing
+ *
+ * @return {Boolean}
+ * @api private
+ */
+
+SendStream.prototype.isFailingPrecondition = function isFailingPrecondition (stat) {
+  // if-match
+  var match = this.req.headers['if-match']
+  var etag = this.res._headers['etag']
+  var strongEtag = /^(?!W\/)/
+
+  if (match) {
+    if (strongEtag.test(etag)) {
+      return !match.split(/ *, */).some(function (match) {
+        return match === '*' || match === etag
+      })
+    }
+  }
+
+  // if-unmodified-since
+  var unmodifiedSince = Date.parse(this.req.headers['if-unmodified-since'])
+  var lastModified = Date.parse(this.res._headers['last-modified'])
```

Comment:
Brought up in #127, we have to remove all uses of `res._headers` property, so best not to add new ones, at least :)